plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/ViperPoweredDeclarationChecker.kt

```diff
@@ -63,10 +53,11 @@ class ViperPoweredDeclarationChecker(private val session: FirSession, private va
             val verifier = Verifier()
 
             val onFailure = { err: VerifierError ->
-                val source = err.position.unwrapOr {
-                    declaration.source
+                val source = err.position.unwrapOr { declaration.source }
+                with(VerifierErrorInterpreter) {
+                    val (diagnosticFactory, message) = err.toHumanReadableMessage()
+                    reporter.reportOn(source, diagnosticFactory, message ?: "", context)
```

Comment:
I am still not convinced that this is the way to go.  Since we have a `KtDiagnosticFactory1` here we should be providing a string, but in the code that is not the case, e.g. `INTERNAL_ERROR` is used without a string being provided, despite it being an important part of the error.